### PR TITLE
Complete use of Maps and Authentication APIs.

### PR DESCRIPTION
**Pull request by alexanderlopez**
Add full functionality of the authentication API in the rsa.html url.

The changes include using the RSA class, implementing the client-server communication to transmit the RSA keys and the cipher to be encrypted/decrypted. The functionality is there, however it still needs styling.
Screenshot 2020-07-12 at 22 26 33
Screenshot 2020-07-12 at 22 26 57

### CODE REVIEW

**Comment by salehhamadeh** [comment]
Since this week is about using the libraries, I wouldn't worry about styling.

File: src/main/java/com/google/sps/data/RSA.java [hunk 1]
```diff
@@ -1,17 +1,73 @@
...
+        n = p.multiply(q);
+        totientN = p.subtract(BigInteger.ONE).multiply(q.subtract(BigInteger.ONE));
+        d = e.modPow(new BigInteger("-1"), totientN);
+    }
+
+    public int getKeySize() {
+        return keySize;
+    }
+
+    public BigInteger encrypt(BigInteger message) {
```
Comment (salehhamadeh) [inline src/main/java/com/google/sps/data/RSA.java]: Are you using the encrypt and decrypt methods anywhere? If not please remove them from this PR and only add them when you use them. Unused methods add noise/confusion to readers.

Comment (alexanderlopez) [reply src/main/java/com/google/sps/data/RSA.java]: Methods removed.

File: src/main/webapp/rsa_script.js [hunk 2]
```diff
@@ -3,9 +3,55 @@ function initTool() {
...
+        document.getElementById('result_text').innerText = result_text;
+    });
+}
+
+function generateKey() {
+    var options = { method: 'POST',
+                    headers: { 'Content-type': 'application/x-www-form-urlencoded'},
+                    body: "action=GENERATE_KEY"};
+    
+    fetch('/rsa-data', options).then(response => response.json()).then(object => {
```
Comment (salehhamadeh) [inline src/main/webapp/rsa_script.js]: nit: object is a generic name. Try to use something more meaningful, like rsaData

**Review by salehhamadeh — APPROVED** [review]

**Pull request merged**